api/core/text2sql_sync.py

```
@@ -0,0 +1,701 @@
+    Raises:
+        Exception: If query fails and cannot be healed.
+    """
+    sanitized_sql, was_modified = auto_quote_sql_identifiers(
+        sql_query, context.known_tables, context.db_type
+    )
+    if was_modified:
+        sql_query = sanitized_sql
+
+    try:
+        query_results = context.loader_class.execute_sql_query(sql_query, context.db_url)
+        return sql_query, query_results
```

Ruff (0.15.5)

[warning] Consider moving this statement to an `else` block

(TRY300)